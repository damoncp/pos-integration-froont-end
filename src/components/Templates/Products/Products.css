.products-subtitle > a {
  font-size: 1rem;
  padding-left: .5rem;
}
