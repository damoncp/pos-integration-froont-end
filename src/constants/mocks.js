const mockProducts = [
  {
    type: "burger",
    data: [
      {
        title: 'Chicken Burger',
        image: 'https://media.istockphoto.com/photos/bacon-burger-picture-id520215281?s=612x612',
        price: 12,
      },
      {
        title: 'Vegan Burger',
        image: 'https://media.istockphoto.com/photos/bacon-burger-picture-id520215281?s=612x612',
        price: 12,
      },
      {
        title: 'Beef Burger',
        image: 'https://media.istockphoto.com/photos/bacon-burger-picture-id520215281?s=612x612',
        price: 12,
      },
      {
        title: 'Lamb Burger',
        image: 'https://media.istockphoto.com/photos/bacon-burger-picture-id520215281?s=612x612',
        price: 12,
      },
    ]
  },
  {
    type: "fries",
    data: [
      {
        title: 'Small Fries',
        image: 'https://media.istockphoto.com/photos/french-fries-with-sauce-picture-id1130883810',
        price: 7,
      },
      {
        title: 'Large Fries',
        image: 'https://media.istockphoto.com/photos/french-fries-with-sauce-picture-id1130883810',
        price: 7,
      },
    ]
  },
  {
    type: "drinks",
    data: [
      {
        title: 'Coca Cola',
        image: 'https://cdn.pixabay.com/photo/2015/01/08/04/18/box-592367_1280.jpg',
        price: 3,
      },
    ]
  },
];

export {
  mockProducts,
};
