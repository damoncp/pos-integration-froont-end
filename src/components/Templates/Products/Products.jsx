import './Products.css';
import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Col, Container, Row, Spinner } from 'reactstrap';
import GlobalContexts from '../../../contexts/globalContexts';
// components
import ProductCard from '../../Atoms/ProductCard/ProductCard';

export default function Products() {
  const {contexts: { products }} = useContext(GlobalContexts);

  return (
    <div className="products">
      <h5 className={"text--centered text--bold heading__default"}>
        Products
      </h5>
      <Container>
        {products.length === 0 ? <Spinner color={"dark"} children="" />
          : products.map(({ type, data: products }, index) => (
            <Row key={`${index}. ${type}`}>
              <h5 className={"text--bold text--capitalize subheading__default"}>
                {type}
              </h5>
              <hr />
              <Row>
                {products.map(({ title, image, price }, index) => (
                  <Col xs={3} key={`${index}. ${title}`}>
                    <ProductCard
                      title={title}
                      image={image}
                      price={price}
                    />
                  </Col>
                ))}
              </Row>
            </Row>
          ))}
      </Container>
    </div>
  )
}

Products.propTypes = {
  text: PropTypes.string,
  classnames: PropTypes.string,
  products: PropTypes.arrayOf({
    type: PropTypes.string,
    data: PropTypes.arrayOf({
      title: PropTypes.string,
      image: PropTypes.string,
      price: PropTypes.number,
    })
  })
};
