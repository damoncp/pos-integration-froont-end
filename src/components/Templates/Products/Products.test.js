import { render, cleanup } from '@testing-library/react';
import Products from './Products';
import GlobalContexts from '../../../contexts/globalContexts';
import { mockProducts } from '../../../constants/mocks';

describe('Test <Products /> component', () => {
  let contexts;
  let renderOrders;

  beforeEach(() => {
    contexts = {
      products: mockProducts,
    };

    renderOrders = render(
      <GlobalContexts.Provider value={{contexts}}>
        <Products />
      </GlobalContexts.Provider>
    );
  });

  afterEach(cleanup);

  it('Products Snapshot test', () => {
    // Arrange
    const { asFragment } = renderOrders;

    // Assert
    expect(asFragment(renderOrders)).toMatchSnapshot();
  });

  it('should render Spinner when products list is empty', () => {
    // Arrange
    contexts = {
      products: [],
    };

    const { container } = render(
      <GlobalContexts.Provider value={{contexts}}>
        <Products />
      </GlobalContexts.Provider>
    );

    // Act
    const spinnerDOM = container.firstChild.lastChild.firstChild;

    // Assert
    expect(spinnerDOM).toHaveClass('spinner-border text-dark')
  });
});
