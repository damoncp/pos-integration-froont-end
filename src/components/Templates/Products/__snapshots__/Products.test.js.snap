// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Products /> component Products Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="products"
  >
    <h5
      class="text--centered text--bold heading__default"
    >
      Products
    </h5>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <h5
          class="text--bold text--capitalize subheading__default"
        >
          burger
        </h5>
        <hr />
        <div
          class="row"
        >
          <div
            class="col-3"
          >
            <div
              class="product-card card"
            >
              <img
                alt="Chicken Burger"
                class="product-card__image card-img-top"
                data-testid="product-card-image"
                src="https://media.istockphoto.com/photos/bacon-burger-picture-id520215281?s=612x612"
              />
              <div
                class="card-body"
              >
                <h5
                  class="product-card__title card-title"
                >
                  <span
                    data-testid="product-card-price"
                  >
                    $12
                  </span>
                   - 
                  <span
                    data-testid="product-card-title"
                  >
                    Chicken Burger
                  </span>
                </h5>
              </div>
            </div>
          </div>
          <div
            class="col-3"
          >
            <div
              class="product-card card"
            >
              <img
                alt="Vegan Burger"
                class="product-card__image card-img-top"
                data-testid="product-card-image"
                src="https://media.istockphoto.com/photos/bacon-burger-picture-id520215281?s=612x612"
              />
              <div
                class="card-body"
              >
                <h5
                  class="product-card__title card-title"
                >
                  <span
                    data-testid="product-card-price"
                  >
                    $12
                  </span>
                   - 
                  <span
                    data-testid="product-card-title"
                  >
                    Vegan Burger
                  </span>
                </h5>
              </div>
            </div>
          </div>
          <div
            class="col-3"
          >
            <div
              class="product-card card"
            >
              <img
                alt="Beef Burger"
                class="product-card__image card-img-top"
                data-testid="product-card-image"
                src="https://media.istockphoto.com/photos/bacon-burger-picture-id520215281?s=612x612"
              />
              <div
                class="card-body"
              >
                <h5
                  class="product-card__title card-title"
                >
                  <span
                    data-testid="product-card-price"
                  >
                    $12
                  </span>
                   - 
                  <span
                    data-testid="product-card-title"
                  >
                    Beef Burger
                  </span>
                </h5>
              </div>
            </div>
          </div>
          <div
            class="col-3"
          >
            <div
              class="product-card card"
            >
              <img
                alt="Lamb Burger"
                class="product-card__image card-img-top"
                data-testid="product-card-image"
                src="https://media.istockphoto.com/photos/bacon-burger-picture-id520215281?s=612x612"
              />
              <div
                class="card-body"
              >
                <h5
                  class="product-card__title card-title"
                >
                  <span
                    data-testid="product-card-price"
                  >
                    $12
                  </span>
                   - 
                  <span
                    data-testid="product-card-title"
                  >
                    Lamb Burger
                  </span>
                </h5>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <h5
          class="text--bold text--capitalize subheading__default"
        >
          fries
        </h5>
        <hr />
        <div
          class="row"
        >
          <div
            class="col-3"
          >
            <div
              class="product-card card"
            >
              <img
                alt="Small Fries"
                class="product-card__image card-img-top"
                data-testid="product-card-image"
                src="https://media.istockphoto.com/photos/french-fries-with-sauce-picture-id1130883810"
              />
              <div
                class="card-body"
              >
                <h5
                  class="product-card__title card-title"
                >
                  <span
                    data-testid="product-card-price"
                  >
                    $7
                  </span>
                   - 
                  <span
                    data-testid="product-card-title"
                  >
                    Small Fries
                  </span>
                </h5>
              </div>
            </div>
          </div>
          <div
            class="col-3"
          >
            <div
              class="product-card card"
            >
              <img
                alt="Large Fries"
                class="product-card__image card-img-top"
                data-testid="product-card-image"
                src="https://media.istockphoto.com/photos/french-fries-with-sauce-picture-id1130883810"
              />
              <div
                class="card-body"
              >
                <h5
                  class="product-card__title card-title"
                >
                  <span
                    data-testid="product-card-price"
                  >
                    $7
                  </span>
                   - 
                  <span
                    data-testid="product-card-title"
                  >
                    Large Fries
                  </span>
                </h5>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <h5
          class="text--bold text--capitalize subheading__default"
        >
          drinks
        </h5>
        <hr />
        <div
          class="row"
        >
          <div
            class="col-3"
          >
            <div
              class="product-card card"
            >
              <img
                alt="Coca Cola"
                class="product-card__image card-img-top"
                data-testid="product-card-image"
                src="https://cdn.pixabay.com/photo/2015/01/08/04/18/box-592367_1280.jpg"
              />
              <div
                class="card-body"
              >
                <h5
                  class="product-card__title card-title"
                >
                  <span
                    data-testid="product-card-price"
                  >
                    $3
                  </span>
                   - 
                  <span
                    data-testid="product-card-title"
                  >
                    Coca Cola
                  </span>
                </h5>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
